src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -11,102 +11,41 @@
 # distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
```

Comment:
Mainly cleanup, refactoring to use new `type_info` functions instead of `TypeInfo` and increased code reuse.